templates/data-provider/index.jsx

```diff
@@ -5,26 +5,36 @@ import styles from './styles.module.css'
 
 import RegistrationLayout from 'modules/registration-layout'
 
-const DataProviderPageTemplate = () => {
-  const handleChange = () => {}
+const DataProviderPageTemplate = ({
+  url,
+  onSubmit,
+  onUrlChange,
+  helperMessage,
+  variant,
+  isFormValid = false,
+}) => {
   const handleSubmit = (event) => {
     event.preventDefault()
-    // API call here probably
+    if (onSubmit) onSubmit(event)
   }
 
   return (
     <RegistrationLayout>
-      <form onChange={handleChange} onSubmit={handleSubmit}>
+      <form onSubmit={handleSubmit}>
         <TextField
           id="data-provider-url"
           type="text"
           name="dataProviderUrl"
           label="Data provider URL"
-          helper="It can be any URL: homepage address, any data resource address, OAI-PMH endpoint or RIOXX endpoint"
+          helper={helperMessage}
           placeholder="For example: oro.open.ac.uk"
+          value={url}
+          onChange={onUrlChange}
+          variant={variant}
+          required
         />
         <div className={styles.submitSection}>
-          <Button variant="contained" disabled>
+          <Button variant="contained" disabled={!isFormValid}>
```

Comment:
I am not sure, where this should be. I see 2 options. The first is listed here, it's how you've done it. The second is like:

```jsx
const FormController = ({ ... }) => {
  const store = useSomethingLikeStore()
  return (
    <Form url={store.query}>
      <Button disabled={!store.query}>Submit</Button>
    </Form>
  )
}
```

So the form is rendered via template but the submit and other actions are added from the controller.

No need to change, keep it in mind 🙂